graphql_compiler/macros/__init__.py

```diff
@@ -1 +1,411 @@
+        # Used as a workaround for GraphQL's lack of support for
+        # inheritance across "types" (i.e. non-interfaces), as well as a
+        # workaround for Gremlin's total lack of inheritance-awareness.
+        # The key-value pairs in the dict specify that the "key" type
+        # is equivalent to the "value" type, i.e. that the GraphQL type or
+        # interface in the key is the most-derived common supertype
+        # of every GraphQL type in the "value" GraphQL union.
+        # Recursive expansion of type equivalence hints is not performed,
+        # and only type-level correctness of this argument is enforced.
+        # See README.md for more details on everything this parameter does.
+        # *****
+        # Be very careful with this option, as bad input here will
+        # lead to incorrect output queries being generated.
+        # *****
+        'type_equivalence_hints',
+
+        # Dict[str, Set[str]] mapping class names to the set of its subclass names.
+        # A class in this context means the name of a GraphQLObjectType,
+        # GraphQLUnionType or GraphQLInterface.
+        'subclass_sets',
+
+        # #################
+        # Macro edge info #
+        # #################
+        # List[MacroEdgeDescriptor] containing all defined macro edges
+        'macro_edges',
+
+        # Dict[str, Dict[str, MacroEdgeDescriptor]] mapping:
+        # class name -> (macro edge name -> MacroEdgeDescriptor)
+        # If a given macro edge is defined on a class X which has subclasses A and B,
+        # then this dict will contain entries for that macro edge for all of [X, A, B].
+        'macro_edges_at_class',
+
+        # Dict[str, Dict[str, MacroEdgeDescriptor]] mapping:
+        # class name -> (macro edge name -> MacroEdgeDescriptor)
+        # If a given macro edge has class X as a target, which has subclasses A and B,
+        # then this dict will contain entries for that macro edge for all of [X, A, B].
+        'macro_edges_to_class',
+
+        # ############################################################
+        # Any other macro types we may add in the future under this. #
+        # ############################################################
+    )
+)
+
+
+def create_macro_registry(schema, type_equivalence_hints=None, subclass_sets=None):
+    """Create and return a new empty macro registry."""
+    if subclass_sets is None:
+        subclass_sets = compute_subclass_sets(schema, type_equivalence_hints=type_equivalence_hints)
+
+    return MacroRegistry(
+        schema_without_macros=schema,
+        type_equivalence_hints=type_equivalence_hints,
+        subclass_sets=subclass_sets,
+        macro_edges=list(),
+        macro_edges_at_class=dict(),
+        macro_edges_to_class=dict())
+
+
+def _find_macro_edge_name_at_subclass(macro_registry, class_name, macro_edge_name):
+    """Return the descriptor for a given macro edge defined on a subclass, if it exists."""
```

Comment:
This implies there's at most one, which is not the case.

Reply:
Would rephrasing to say `Return any descriptor` resolve the concern, or would you prefer something else?